crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -300,4 +303,107 @@ impl BeaconState {
+        if index as usize >= self.validators.len() {
+            return;
+        }
+        if self.validators.get(index as usize).unwrap().exit_epoch != FAR_FUTURE_EPOCH {
+            return;
+        }
+
+        let mut exit_epochs: Vec<u64> = self
+            .validators
+            .iter()
+            .filter_map(|v| {
+                if v.exit_epoch != FAR_FUTURE_EPOCH {
+                    Some(v.exit_epoch)
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        exit_epochs.push(compute_activation_exit_epoch(self.get_current_epoch()));
+        let mut exit_queue_epoch = *exit_epochs.iter().max().unwrap_or(&0);
+
+        let exit_queue_churn = self
+            .validators
+            .iter()
+            .filter(|v| v.exit_epoch == exit_queue_epoch)
+            .count();
+
+        if exit_queue_churn >= self.get_validator_churn_limit() as usize {
+            exit_queue_epoch += 1;
+        }
+
+        // Set validator exit epoch and withdrawable epoch
+        if let Some(validator) = self.validators.get_mut(index as usize) {
+            validator.exit_epoch = exit_queue_epoch;
+            validator.withdrawable_epoch =
+                validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY;
+        }
+    }
+
+    /// Slash the validator with index ``slashed_index``
+    pub fn slash_validator(
+        &mut self,
+        slashed_index: u64,
+        whistleblower_index: Option<u64>,
+    ) -> anyhow::Result<()> {
+        let epoch = self.get_current_epoch();
+
+        // Initiate validator exit
+        self.initiate_validator_exit(slashed_index);
+
+        // Mutably borrow the slashed validator
+        let mut binding = self.clone();
+        let validator = binding
+            .validators
+            .get_mut(slashed_index as usize)
+            .ok_or_else(|| anyhow::anyhow!("Validator at index {} not found", slashed_index))?;
+
+        validator.slashed = true;
+        validator.withdrawable_epoch = std::cmp::max(
+            validator.withdrawable_epoch,
+            epoch + EPOCHS_PER_SLASHINGS_VECTOR,
+        );
+
+        // Add slashed effective balance to the slashings vector
+        self.slashings[(epoch % EPOCHS_PER_SLASHINGS_VECTOR) as usize] +=
+            validator.effective_balance;
+
+        // Decrease validator balance
+        self.clone().decrease_balance(
+            slashed_index,
+            validator.effective_balance / MIN_SLASHING_PENALTY_QUOTIENT,
+        );
+
+        // Apply proposer and whistleblower rewards
+        let proposer_index = self.get_beacon_proposer_index()?;
+        let whistleblower_index = whistleblower_index.unwrap_or(proposer_index);
+
+        let whistleblower_reward = validator.effective_balance / WHISTLEBLOWER_REWARD_QUOTIENT;
+        let proposer_reward = (whistleblower_reward * PROPOSER_WEIGHT) / WEIGHT_DENOMINATOR;
```

Comment:
```suggestion
        let proposer_reward = whistleblower_reward * PROPOSER_WEIGHT / WEIGHT_DENOMINATOR;
```

these brackets aren't needed due to order of operations